### DISCUSSION TITLE
Error: /user/installations/{installation_id}/repositories

### DISCUSSION THREAD

**Discussion by zegl** [community/community#35412] [Apps, API and Webhooks] [answered]
Hey Community (and GitHub staff!)

New requests to the `/user/installations/{installation_id}/repositories` endpoint (using a "User-to-Server" OAuth2 token) are now failing with a 403 status code and the following message:

>  You must authenticate with an access token authorized to a GitHub App, a personal access token, or basic auth in order to list repositories for an installation.

The [documentation](https://docs.github.com/en/rest/apps/installations#list-repositories-accessible-to-the-user-access-token) for said endpoint states that the access token **must** be a "User-to-Server" token, which makes sense given the functionality of the endpoint.

Has anyone heard anything about a planned change to this endpoint? The same code has been working for multiple months on our side, and has suddenly stopped working without any changes to the logic on our side.

Notice the discrepancy between the error message and the documentation. This indicates that this might be an issue on GitHubs side.

Other requests, such as requests to `/user/installations` are working as usual with the same "User-to-Server" token as later used to the `/user/installations/{installation_id}/repositories` endpoint.

---

I've also raised a GitHub ticked (`#1823707`) about this same issue. I'm starting this discussion in hopes notifying others that are hitting the same issue, and to get the attention of someone at GitHub.

**Comment by prabenzom** [comment]
We are seeing this issue as well

**Comment by zegl** [comment, marked as answer]
I got an answer from my private ticket, sharing the answer here for visibility:


> Thanks for reaching out to GitHub Support!
> 
> It looks like there was a recent change that triggered that behavior. From my tests, this error happens in the case where User A is trying to list repos available to them in User B -- this affects only non-organization installations. In this case, User B has both installed the app on their user scope and has also granted User A collaborator permissions on at least one repository.
> 
> I have filed an internal issue with the engineering team to ask if this behavior is intended and what > the current expectation is like for the endpoint. I don't have an ETA for a response now but write to you as soon as I have any news.

**Comment by prabenzom** [comment]
Thank you!
